streams/src/main/java/org/apache/kafka/streams/errors/TaskMigratedException.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.errors;
+
+
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.streams.processor.TaskId;
+
+/**
+ * Indicates that a task got migrated to another thread.
+ * Thus, the task raising this exception can be cleaned up and closed as "zombie".
+ */
+public class TaskMigratedException extends StreamsException {
+
+    private final static long serialVersionUID = 1L;
+
+    public TaskMigratedException(final TaskId taskId) {
+        this(taskId, null);
+    }
+
+    public TaskMigratedException(final TaskId taskId,
```

Comment:
Could we include not only the `TaskId` but the `Task` and print `task.toString` in the error message to show which stores are correlated to the changelog topics as well? Some related comments are below.